### DISCUSSION TITLE
Brightness Transition API or Fade-in/Fade-out Effect

### DISCUSSION THREAD

**Discussion by acortelyou** [awawa-dev/HyperHDR#917] [Ideas]
I'm using Home Assistant to synchronize the on/off state of the backlight with the TV as the video source (roku/shield-type box) never actually shuts off.

This works but it's a bit jarring when the system turns on and off.

Sending separate brightness commands is jerky.

Is it possible the current API to give a transition time along with the brightness command? As many other "smart lights" have?

I don't see anything like that in the code at the moment but that seems like the simplest to implement.

Alternatively, is it possible for an effect to get the current grabber frame that would be displayed if the effect wasn't on top?

If so, could write an effect which modifies the current image to achieve something fun like [a retro signal fade/shrink vertical/grow horizontal CRT effect](https://www.pond5.com/stock-footage/item/130430252-effect-turning-tv-and-no-signal-old-tv-error-video-damage). 

Food for thought, let me know what you think the best way to achieve this would be.

**Comment by awawa-dev** [comment]
Hi
There is currently no such solution. There are some fade-in/out effects, but they are parameterless. Theoretically it is possible to provide the current captured grabber frame for the effect, but it will require some minor changes to get it (~~it sits in the muxer object related to the instance~~ better to use GlobalSignals->SignalNewVideoImage to capture it first). And of course writing new animation in c++ based on this frame.
